src/couch_replicator/src/couch_replicator_api_wrap.erl

```diff
@@ -59,11 +59,11 @@
 db_uri(#httpdb{url = Url}) ->
     couch_util:url_strip_password(Url);
 
-db_uri(#db{name = Name}) ->
-    db_uri(Name);
+db_uri(DbName) when is_binary(DbName) ->
```

Comment:
should we add a clause with `when is_list(DbName)` guard? 
Is there any possibility that `couch_db:name` would return list instead of binary (for example for fake_db or something). 
In such case the current implementation would crash with badmatch.